api/tests/opentrons/protocol_engine/state/test_command_store.py

```diff
@@ -527,13 +536,44 @@ def test_command_store_ignores_known_finish_error() -> None:
     assert subject.state == CommandState(
         queue_status=QueueStatus.INACTIVE,
         run_result=RunResult.FAILED,
-        is_hardware_stopped=False,
+        run_completed_at=None,
+        is_door_blocking=False,
+        running_command_id=None,
+        all_command_ids=[],
+        queued_command_ids=OrderedSet(),
+        commands_by_id=OrderedDict(),
+        errors_by_id={},
+        run_started_at=None,
+    )
+
+
+def test_command_store_save_started_completed_run_timestamp() -> None:
+    """Should return a none empty run_completed_at and run_started_at."""
+    subject = CommandStore()
+    start_time = datetime.now()
+    hardware_stopped_time = datetime.now()
+    resume_time = datetime.now()
+    second_resume_time = datetime.now()
```

Comment:
Same comment as above. In addition introducing randomness of input data into the test, which makes assertion failures harder to read when it happens, the accuracy of the test depends on two subsequent calls to `datetime.now()` returning different datetimes.

We should construct distinct values ourselves to ensure the test is giving us the value we need.

```suggestion
    start_time = datetime(year=2021, month=1, day=1)
    hardware_stopped_time = datetime(year=2022, month=2, day=2)
    resume_time = datetime(year=2023, month=3, day=3)
    second_resume_time = datetime(year=2024, month=4, day=4)
```